docs/dev-guide/plugin-dev.md

```diff
@@ -296,41 +423,397 @@ module.exports = api => {
+
+```js
+api.describeConfig({
+  // other config properties
+  onRead: ({ data }) => ({
+    prompts: [
+      {
+        name: `color`,
+        type: 'input',
+        message: 'Define the color for greeting message',
+        default: 'white'
+      }
+    ]
+  }),
+});
+```
+
+In the example above we specified the input prompt with the default value of 'white'. This is how our configuration screen will look with all the settings provided above:
+
+![UI Config Start](/ui-config-start.png)
+
+Now let's replace hardcoded `white` value with the property from the config file. In the `onRead` hook `data` object contains the JSON result of each config file content. In our case, the content of `myConfig.js` was
+
+```js
+//myConfig.js
+
+module.exports = {
+  color: 'black',
+};
+```
+
+So, the `data` object will be
+
+```js
+{
+  // File
+  myConfig: {
+    // File data
+    color: 'black'
+  }
+}
+```
+
+It's easy to see that we need `data.myConfig.color` property. Let's change `onRead` hook:
+
+```js
+onRead: ({ data }) => ({
+  prompts: [
+    {
+      name: `color`,
+      type: 'input',
+      message: 'Define the color for greeting message',
+      default: data.myConfig.color
+    }
+  ]
+}),
+```
+
+You can see that on the configuration screen `white` is replaced with `black`.
+
+#### Save config changes
+
+We just read the content of `myConfig.js` and used it on the configuration screen. Now let's try to save any changes done in the color input field to the file. We can do it with the `onWrite` hook:
+
+```js
+api.describeConfig({
+  /* ... */
+  onWrite: ({ prompts, api }) => {
+    // ...
+  }
+})
+```
+
+`onWrite` hook can take a lot of [arguments](ui-api.html#save-config-changes) but we will need only two of them: `prompts` and `api`. First one is current prompts runtime objects - we will get a prompt id from it and retrieve an answer with this id. To retrieve the answer we'll use `async getAnswer()` method from the `api`:
+
+```js
+async onWrite({ api, prompts }) {
+  const result = {};
+  for (const prompt of prompts) {
+    result[`${prompt.id}`] = await api.getAnswer(prompt.id);
+  }
+  api.setData('myConfig', result);
+}
+```
+
+Now if you try to change the value in the color input field from `black` to `red` on the config screen and press `Save the changes`, you will observe that `myConfig.js` file in your project has been changed as well:
+
+```js
+//myConfig.js
+
+module.exports = {
+  color: 'red',
+};
+```
+
+### Display prompts
+
+If you want, you can display [prompts](#prompts) in the Vue UI as well. If you will install your plugin through the UI, prompts will be shown on the plugin invocation step.
+
+To display a prompt in the UI you need to extend the [inquirer object](#prompts-for-3rd-party-plugins) with additional properties. They are optional and only used by the UI:
```

Comment:
Actually you don't have to. Prompts will still be displayed anyway. However you can "augment" them for the UI with the UI-specific properties.